File: Controls/FilePickerMaterialCombineJava/app/src/main/java/com/kelfan/filepicker/FileUtils.java
package com.kelfan.filepicker;

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileFilter;
import java.io.FileOutputStream;
import java.io.OutputStreamWriter;
import java.io.Writer;
import java.text.DecimalFormat;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

/**
 * Created by Administrator on 7/03/2018.
 */

public class FileUtils {
    static List<File> getFileListByDirPath(String path, FileFilter filter) {
        File directory = new File(path);
        File[] files = directory.listFiles(filter);

        if (files == null) {
            return new ArrayList<>();
        }

        List<File> result = Arrays.asList(files);
        Collections.sort(result, new FileComparator());
        return result;
    }

    static String cutLastSegmentOfPath(String path) {
        if (path.length() - path.replace("/", "").length() <= 1)
            return "/";
        String newPath = path.substring(0, path.lastIndexOf("/"));
        // We don't need to list the content of /storage/emulated
        if (newPath.equals("/storage/emulated"))
            newPath = "/storage";
        return newPath;
    }

    static String getReadableFileSize(long size) {
        if (size <= 0) return "0";
        final String[] units = new String[]{"B", "KB", "MB", "GB", "TB"};
        int digitGroups = (int) (Math.log10(size) / Math.log10(1024));
        return new DecimalFormat("#").format(size / Math.pow(1024, digitGroups)) + " " + units[digitGroups];
    }

    static void createFile(String filename) {
        File file = new File(filename);
        if (!file.exists()) {
            try {
                Writer myWriter = new BufferedWriter(new OutputStreamWriter(
                        new FileOutputStream(filename), "UTF-8"));
                myWriter.write("");
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }

    static void createFolder(String filename) {
        File folder = new File(filename);
        if (!folder.exists()) {
            folder.mkdirs();
        }
    }

}

File: Controls/FilePickerMaterialKotlin/app/src/main/java/com/example/jzheng4/filepickermaterialkotlin/utils/FileUtils.kt
package com.example.jzheng4.filepickermaterialkotlin.utils

import java.io.File
import java.io.FileFilter
import java.text.DecimalFormat
import java.util.*

/**
 * Created by jzheng4 on 20/02/2018.
 */
object FileUtils {
    fun getFileListByDirPath(path: String, filter: FileFilter): List<File> {
        val directory = File(path)
        val files = directory.listFiles(filter) ?: return ArrayList()

        val result = Arrays.asList(*files)
        Collections.sort(result, FileComparator())
        return result
    }

    fun cutLastSegmentOfPath(path: String): String {
        if (path.length - path.replace("/", "").length <= 1)
            return "/"
        var newPath = path.substring(0, path.lastIndexOf("/"))
        // We don't need to list the content of /storage/emulated
        if (newPath == "/storage/emulated")
            newPath = "/storage"
        return newPath
    }

    fun getReadableFileSize(size: Long): String {
        if (size <= 0) return "0"
        val units = arrayOf("B", "KB", "MB", "GB", "TB")
        val digitGroups = (Math.log10(size.toDouble()) / Math.log10(1024.0)).toInt()
        return DecimalFormat("#").format(size / Math.pow(1024.0, digitGroups.toDouble())) + " " + units[digitGroups]
    }
}